### ISSUE TITLE
[tar_stream] time is being rounded, exact nanoseconds aren't being preserved

### ISSUE THREAD

**Issue by stephen679** [vbatts/go-mtree#26]
basic `tar -cf` does not seem to be preserving exact mod time in nanoseconds. Below is output of creating a new mtree spec from extracted tar file.
```
[schung@localhost Desktop] $ gomtree -c -p newdemo
#          user: schung
#       machine: localhost.localdomain
#          tree: newdemo
#          date: Mon Jul 18 10:51:07 2016

# .
/set type=file nlink=1 mode=0664 uid=1000 gid=1000
. size=4096 type=dir mode=0775 nlink=2 time=1468853428.512943631
    file1 size=0 time=1468853425.971947758
    file2 size=0 time=1468853428.512943631
..

```
But:
```
[schung@localhost Desktop] $ tar -cf newdemo.tar newdemo
[schung@localhost Desktop] $ tar --same-owner -xvf newdemo.tar
newdemo/
newdemo/file1
newdemo/file2
[schung@localhost Desktop] $ gomtree -c -p newdemo
#          user: schung
#       machine: localhost.localdomain
#          tree: newdemo
#          date: Mon Jul 18 10:51:43 2016

# .
/set type=file nlink=1 mode=0664 uid=1000 gid=1000
. size=4096 type=dir mode=0775 nlink=2 time=1468853428.000000000
    file1 size=0 time=1468853425.000000000
    file2 size=0 time=1468853428.000000000
..
```

**Comment by vbatts** [comment]
yeah, interesting. This is a limitation of the size of the field in the tar archive.
https://gist.github.com/vbatts/75320e0384bf9e72a6eff219721e07c9
renders the output:
```
1468952103911592902
1468952103911665759
1468952103911693524
Contents of readme.txt (%!q(int64=1468952103000000000)):
This archive contains some text files.
Contents of gopher.txt (%!q(int64=1468952103000000000)):
Gopher names:
George
Geoffrey
Gonzo
Contents of todo.txt (%!q(int64=1468952103000000000)):
Get animal handling license.
```

**Comment by stephen679** [comment]
Resolved with PR #29 implementing an extra `tar_time` keyword.
